lms/views/legacy_content_item_selection.py

```diff
@@ -0,0 +1,90 @@
+from __future__ import unicode_literals
+
+from pyramid.view import view_config
+
+from lms.exceptions import MissingLTIContentItemParamError
+from lms.util import lti_launch
+from lms.util.view_renderer import view_renderer
+from lms.util.associate_user import associate_user
+from lms.util.authorize_lms import authorize_lms
+from lms.views.decorators import legacy_upsert_h_user
+from lms.views.decorators import legacy_create_course_group
+from lms.views.helpers import canvas_files_available
+
+
+@view_config(route_name="content_item_selection", request_method="POST")
+@lti_launch
+@legacy_upsert_h_user
+@legacy_create_course_group
+@associate_user
+@authorize_lms(
+    authorization_base_endpoint="login/oauth2/auth",
+    redirect_endpoint="canvas_oauth_callback",
+    oauth_condition=canvas_files_available,
+)
+def content_item_selection(request, _jwt, **_):
+    """
+    Render the form that teachers see to configure the module item.
+
+    This view is only used for lms's that support link selection
+    """
+    lms_url = request.find_service(name="ai_getter").lms_url(
+        request.params["oauth_consumer_key"]
+    )
+    return content_item_form(
+        request,
+        lti_params=request.params,
+        lms_url=lms_url,
+        content_item_return_url=request.params["content_item_return_url"],
+        jwt=None,
+    )
+
+
+@view_renderer(
+    renderer="lms:templates/content_item_selection/new_content_item_selection.html.jinja2"
+)
+def content_item_form(request, lti_params, lms_url, content_item_return_url, jwt=None):
+    for param in ["lti_version", "oauth_version", "oauth_nonce", "oauth_signature"]:
+        if param not in lti_params:
+            raise MissingLTIContentItemParamError(param)
+
+    form_fields = {
+        "context_id": lti_params["context_id"],
+        "user_id": lti_params["user_id"],
+        "lti_message_type": "ContentItemSelection",
+        "lti_version": lti_params["lti_version"],
+        "oauth_version": lti_params["oauth_version"],
+        "oauth_nonce": lti_params["oauth_nonce"],
+        "oauth_consumer_key": lti_params["oauth_consumer_key"],
+        "oauth_signature_method": lti_params["oauth_signature_method"],
+        "oauth_signature": lti_params["oauth_signature"],
+        "jwt_token": jwt,
+    }
+    # These fields appear in blackboard launches, but not in canvas
+    # launches
+    if "resource_link_id" in lti_params:
```

Comment:
What is going to happen with the Blackboard-specific logic here in future?

Reply:
I think that comment is misleading. These params aren't just for Blackboard, they're for all LMS's that don't use content item selection, which is all but Canvas. I think Blackboard was just the first non-content item selection LMS that support was added for and so various code comments refer to "Blackboard".

I think what has happened here is that this legacy view code was used for both LMS's that support content item selection (Canvas) and for doing unconfigured assignment launches in LMS's that don't support it. With the feature-flagged code we've already moved those unconfigured assignment launches, which includes Blackboard, to the new `unconfigured_basic_lti_launch()` view. So if Blackboard is gonna break, we've already broken it!

I don't think it is broken though because the new `unconfigured_basic_lti_launch()` does retain support for these "Blackboard-specific" params.

Our Blackboard test site was down. It's back up now, but it doesn't work with dev environments, so I'll have to deploy to QA to test it